framework/DataObjects/DataSet.py

```diff
@@ -0,0 +1,1853 @@
+    elif variable is not None:
+      # remove from self._data
+      if not noData:
+        self._data = self._data.drop(variable)
+      # remove from self._collector
+      if not noColl:
+        varIndex = self._orderedVars.index(variable)
+        self._collector.removeEntity(varIndex)
+      # remove references to variable in lists
+      self._orderedVars.remove(variable)
+      # TODO potentially slow lookups
+      for varlist in [self._inputs,self._outputs,self._metavars]:
+        if variable in varlist:
+          varlist.remove(variable)
+      # remove from pivotParams, and remove any indexes without keys
+      for pivot in self.indexes:
+        if variable in self._pivotParams[pivot]:
+          self._pivotParams[pivot].remove(variable)
+        if len(self._pivotParams[pivot]) == 0:
+          del self._pivotParams[pivot]
+          # if in self._data, clear the index
+          if not noData and pivot in self._data.dims:
+            del self._data[pivot] # = self._data.drop(pivot,dim
+      # TODO remove references from general metadata?
+    if self._scaleFactors is not None:
+      self._scaleFactors.pop(variable,None)
+    #either way reset kdtree
+    self.inputKDTree = None
+
+  def renameVariable(self,old,new):
+    """
+      Changes the name of a variable from "old" to "new".
+      @ In, old, str, old name
+      @ In, new, str, new name
+      @ Out, None
+    """
+    # determine where the old variable was
+    isInput = old in self._inputs
+    isOutput = old in self._outputs
+    isMeta = old in self._metavars
+    isIndex = old in self.indexes
+    # make the changes to the variable listings
+    if isInput:
+      self._inputs = list(a if (a != old) else new for a in self._inputs)
+    if isOutput:
+      self._outputs = list(a if (a != old) else new for a in self._outputs)
+    if isMeta:
+      self._metavars = list(a if (a != old) else new for a in self._metavars)
+    if isIndex:
+      # change the pivotParameters listing, as well as the sync/unsynced listings
+      self._pivotParams[new] = self._pivotParams.pop(old)
+      if old in self._alignedIndexes.keys():
+        self._alignedIndexes[new] = self._alignedIndexes.pop(old)
+      else:
+        self._orderedVars = list(a if a != old else new for a in self._orderedVars)
+    # if in/out/meta, change allvars (TODO wastefully already done if an unaligned index)
+    if isInput or isOutput or isMeta:
+      self._orderedVars = list(a if a != old else new for a in self._orderedVars)
+    # change scaling factor entry
+    if old in self._scaleFactors:
+      self._scaleFactors[new] = self._scaleFactors.pop(old)
+    if self._data is not None:
+      self._data.rename({old:new},inplace=True)
+
+  def reset(self):
+    """
+      Sets this object back to its initial state.
+      @ In, None
+      @ Out, None
+    """
+    self._data = None
+    self._collector = None
+    self._meta = {}
+    # TODO others?
```

Comment:
I think we left those three, because we aren't completely gutting the data object, just removing the data contents; the same variables should still exist.  I'll note that in the docstrings.